src/js/utils/language.js

```diff
@@ -61,11 +63,41 @@ export function getLanguage() {
 export const translatedLanguageCodes = ['ar', 'da', 'de', 'es', 'fr', 'it', 'ja', 'nb', 'nl', 'pt', 'ro', 'sv', 'tr', 'zh'];
 
 export function isTranslationAvailable(language) {
-    // 'en' is our default language, therefore it is excluded from the translated language codes.
     return translatedLanguageCodes.indexOf(formatLanguageCode(language)) >= 0;
 }
 
-export function loadJsonTranslation(base, languageCode, oncomplete, onerror) {
-    const url = base + 'translations/' + languageCode + '.json';
-    ajax({ url, oncomplete, onerror, responseType: 'json' });
+export function getCustomLocalization(localization, intl, languageAndCountryCode) {
+    localization = localization || {};
+    intl = intl || {};
+    const languageCode = formatLanguageCode(languageAndCountryCode);
+    const languageAndCountryCustomization = languageCode === languageAndCountryCode ? {} :
+        intl[languageAndCountryCode] || intl[languageAndCountryCode.toLowerCase()] ||
+        intl[languageAndCountryCode.replace('-', '_')] || intl[languageAndCountryCode.toLowerCase().replace('-', '_')];
+        // TODO: Country codes are generally seen in upper case, but we have yet to find documentation enforcing this format.
+        // When the documentation is found, remove lower case support and update our docs with reference to standards.
+    return Object.assign({}, localization, intl[languageCode], languageAndCountryCustomization);
+}
+
+export function isLocalizationComplete(customLocalization) {
+    return isDeepKeyMatch(en, customLocalization);
+}
+
+export function loadJsonTranslation(base, languageCode) {
+    const url = `${base}translations/${formatLanguageCode(languageCode)}.json`;
+    return new Promise((resolve, reject) => {
+        const oncomplete = (result) => resolve(result);
+        const onerror = () => reject();
+        ajax({ url, oncomplete, onerror, responseType: 'json' });
+    });
+}
+
+export function applyTranslation(translationJson, customization) {
+    translationJson = translationJson || {};
+    const localization = Object.assign({}, en, translationJson, customization);
+    localization.errors = Object.assign({}, en.errors, translationJson.errors, customization.errors);
+    localization.related = Object.assign({}, en.related, translationJson.related, customization.related);
+    localization.sharing = Object.assign({}, en.sharing, translationJson.sharing, customization.sharing);
+    localization.advertising = Object.assign({}, en.advertising, translationJson.advertising, customization.advertising);
```

Comment:
Can you make this DRYer?

This is 100 bytes smaller minified https://babeljs.io/repl/#?babili=true&browsers=&build=&builtIns=false&spec=false&loose=true&code_lz=GYVwdgxgLglg9mABAQwA6oDYE8AqAnZMAZw2VgQAooDjTywApIhAGkQhCKjgFsYAvMvDABKRAG8AUIhnsEXRBjgRkGAUISIAvIgDyAIwBWAU2gA6ZESIwA5mAriAvm2Ng21QiQ2Nmb9p24-QXoRAG5pWR5jPBtjCiUVNWDhNgByaLw4PCJUlz8PWm8mVn8uXnUQ8NlEKJi4hNUKlMRUvGM6YwATXMRXdxovemK_DjKg7zCImVrY-OVG5JLUogALZDwYMBsevsQCweFhtlHApoRJ6pn6-aTvNOROgDdo2GstnfyBukPfY4DyxaiKqINpQEB4JANW70cKOSSSUCQeg1aKzfhsVCZVBsZBsfTHMRSGT8ADamLgqAAuto9EZTFALFZbPYnDiyVjKXj2RTOexuVTJo4gA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Cbabili%2Cenv&prettier=true&targets=&version=6.26.0&envVersion=1.6.2